zipline/finance/blotter.py

```diff
@@ -136,6 +140,22 @@ def cancel(self, order_id):
             # along with newly placed orders.
             self.new_orders.append(cur_order)
 
+    def execute_cancel_policy(self, event):
+        if self.cancel_policy.should_cancel(event):
+            for orders_by_sid in itervalues(self.open_orders):
+                for order in orders_by_sid:
+                    self.cancel(order.id)
+                    if self.cancel_policy.warn_on_cancel:
```

Comment:
`self.cancel_policy.warn_on_cancel` can't change during these loops. can you pull the check out of the loops, maybe right after the `should_cancel` check?